common/djangoapps/student/tests/test_retirement.py

```diff
@@ -248,6 +238,11 @@ class TestRegisterRetiredUsername(TestCase):
     """
     Tests to ensure that retired usernames can no longer be used in registering new accounts.
     """
+    # The returned message here varies depending on whether a ValidationError -or-
+    # an AccountValidationError occurs.
+    START_END_MSG_1 = ('An account with the Public Username', 'already exists.')
+    START_END_MSG_2 = ('It looks like', 'belongs to an existing account. Try again with a different username.')
```

Comment:
Here's the main difference between the reverted PR and this one. Allow for different returned error messages depending on which error occurs.